flake.nix

```diff
@@ -1,21 +1,25 @@
 {
   description = "nixos-generators - one config, multiple formats";
 
+  # Lib dependency
+  inputs.nixlib.url = "github:divnix/nixpkgs.lib";
```

Comment:
I have seen this project before but it ist not be easier to make the `nixpkgs` input overridable and than override nixpkgs as a whole instead of having a lib input and a nixpkgs input? Nixpkgs is required anyway so you are just adding more dependencies? What I would understand is if the generators project would ouput library functions that can be used without nixpkgs but right now it only outputs a package, which depends on nixpkgs.
Please clarify further.

Reply:
So what do you mean by instantiating? The only thing you can instantiate from here is the package itself, which needs nixpkgs.

Reply:
I think I understand it now. You are using this lib thing to iterate over all modules and than use nixpkgs later when evaluating nixpkgs. Do you have automation to keep this nixpkgs lib fork automatically up-to-date?

Reply:
> I think I understand it now. You are using this lib thing to iterate over all modules and than use nixpkgs later when evaluating nixpkgs.

Correct!

> Do you have automation to keep this nixpkgs lib fork automatically up-to-date?

[Yes :smile: -- every sunday](https://github.com/divnix/nixpkgs.lib/blob/master/.github/workflows/update.yml#L6)

Reply:
A little more context:

- Technically, there is a little overhead: way downstream nixpkgs would _also_ instantiate `lib` (and that would or would not be used).
- Practically, downstream has a lot less change to depend (by mishap) of two slightly different versions of nixpkgs **just because of** the lib dependency . If that were to happen, it would be costly (2 complete nixpkgs evaluations).

Note, that all `nixpkgs` transiently appear in `flake.lock`, but as long as they are not accesored, they are neither fetched (sic!) nor instantiated.